### PR TITLE
Allow users to create ereaders

### PR DESCRIPTION
**Pull request by laustindasauce**
# Allow users to create ereaders

This is an enhancement feature that should close issue #1982 when merged. The purpose is to allow self-serve of the ereader devices by users of an Audiobookshelf instance. Personally, I'd much rather pass on the handling of ereader devices to the users themselves. Since it's disabled by default, it won't have any effect on current instances and only those who would like to utilize the feature can now have the opportunity to.

### Client

- Created label for new permission with value 'Can Create Ereader' 
- Added toggle to Account modal for user update/creation
- For non admin/root, added ereader table to account page above logout button
    - Only allows edit/delete options for ereaders solely accessible by the user
- Created new user ereader modal for creating/updating ereaders at user level
    - Removes accessibility options from the form (only name & email can be changed)
    - Only passes ereader devices solely accessible by the user to modal
- Toast for duplicate name error when found by server

### Server

- Added `createEreader` permission for users with default false except for admin / root
- Created new API route - POST: `/api/me/ereader-devices`
- New MeController method to handle the route
    - Handles check for duplicate names across ereader devices not owned by the user
    - Only allows update/deletion/creation of devices with the current user as the **only** accessible user

## Screenshots

User account page with createEreader permission

Admin users page with new toggle

### CODE REVIEW

File: client/strings/en-us.json [hunk 1]
```diff
@@ -456,6 +456,7 @@
   "LabelPermissionsAccessAllLibraries": "Can Access All Libraries",
   "LabelPermissionsAccessAllTags": "Can Access All Tags",
   "LabelPermissionsAccessExplicitContent": "Can Access Explicit Content",
```
[the remark this answers is not in the document]
Comment (laustindasauce) [reply client/strings/en-us.json:458]: Yep, no problem. Wasn't sure how that worked.

**Comment by nichwall** [comment]
This is generally working well for me and the correct devices show up for the user with limited editing if they are not the sole user who can access it. I did find some problems using the following tests.

Test 1:
- Log in as user who can edit e-reader devices
- Create device
- Observe new device is present
- Navigate to home page
- Navigate to user page
- Observe device is missing
- Refresh page (using F5), device is now visible

Test 2:
- Create 2 devices. One is only accessible to a specific user, the other is available for all users
- Log in as the specific user
- Add a device
- The second device (available to all users) is overwritten in the table
- Logging back in to root user shows that the second device is missing and was replaced by the newly created device

**Comment by laustindasauce** [comment]
I was able to reproduce both issues and actually found another while doing that. These issues happened because I last minute decided to remove shared ereaders from the payload so that if a user uses the API they can't be a bad actor/unintentionally update any shared ereaders.

1. Issue with deleting shared ereader
Resolved by fixing the logic in how server side retrieves 'otherDevices' with [this change](https://github.com/advplyr/audiobookshelf/pull/3531/commits/9ebed5e5133b6a2ef0853abe8bd4bdc132b03b49#diff-5770d745d750e9b33e8f763d2c2bb2ad9a5a1ef8c8952e32031b905e907cf57bR416)

2. Issue with changes not being pushed properly (forced reload)
Resolved by fixing the how the clientEmitter is called for socket changes with [this change](https://github.com/advplyr/audiobookshelf/pull/3531/commits/9ebed5e5133b6a2ef0853abe8bd4bdc132b03b49#diff-5770d745d750e9b33e8f763d2c2bb2ad9a5a1ef8c8952e32031b905e907cf57bR437)

3. Issue with inability to delete user owned ereaders
- Would throw duplicate name error since we were including the shared ereaders in the API call
Resolved by only passing the user owned ereaders with [this change](https://github.com/advplyr/audiobookshelf/pull/3531/commits/1fb17ef1feaa76dc49803ca0fcb9ab4c0942c72c)

**Comment by advplyr** [comment]
Solid, thanks!

**Pull request merged**